### DISCUSSION TITLE
Comparison between pXML and Ren

### DISCUSSION THREAD

**Discussion by celtic-coder** [pxml-lang/pxml-lang.github.io#2] [Ideas]
Hi Christian (@pxml-lang),

Having dabbled with Rebol / Red in the long-distant past, I came across [Ren - The Human Notation](https://pointillistic.com/ren/).

Given that Rebol has over [fifty different data types](http://rebol.com/r3/docs/datatypes.html), it leans towards capturing different data type information.

However, despite this leaning, there appears to be a measure of overlap between pXML and Ren. Might there be some ideas that could be incorporated into pXML, perhaps? The [About](https://pointillistic.com/ren/ren-about.html) and [FAQ](https://pointillistic.com/ren/ren-faq.html) pages explain the background and philosophy of the notation.

Kind Regards,
Liam

**Comment by pxml-lang** [comment]
Thanks for these interesting links. Will have a thorough look at them. I know Rebol and Red, but never heard of Ren.

As for 'types in pXML' I think that basic pXML should only support strings, exactly as in XML.

I have some ideas as to how types (as well as other features) could be added as _optional extensions_ to pXML. Besides basic, common, predefined types (booleans, numbers, dates, lists, maps, sets, etc.) it must be easy to define new customized types with string parsers and validation logic. But I need more time for pondering and writing some proof-of-concept code before publishing and discussing such extensions.

In the meantime, any ideas and discussions in this context are very welcome.

(Note: I moved this item from 'Issues' to 'Discussions' because the subject is about possible future extensions.)
